apps/frontend/core-dapp/src/features/trade/close-trade/CloseTradeStore.ts

```diff
@@ -0,0 +1,56 @@
+import { BigNumber } from 'ethers'
+import { makeAutoObservable } from 'mobx'
+import { validateStringToBN } from 'prepo-utils'
+import { RootStore } from '../../../stores/RootStore'
+import { MarketEntity } from '../../../stores/entities/MarketEntity'
+import { PositionEntity } from '../../../stores/entities/Position.entity'
+
+export class CloseTradeStore {
+  input: { type: 'max' } | { type: 'input'; value: string } = {
+    type: 'input',
+    value: '',
+  }
+  constructor(private root: RootStore) {
+    makeAutoObservable(this, {}, { autoBind: true })
+  }
+
+  // selected market and positions, for ease of accessing
+  // so we don't have to keep doing this.root.tradeStore...
+  // and can just use this.selectedMarket
+  get selectedMarket(): MarketEntity | undefined {
+    return this.root.tradeStore.selectedMarket
+  }
+
+  get selectedPosition(): PositionEntity | undefined {
+    return this.root.tradeStore.selectedPosition
+  }
+
+  // input states
+
+  setInput(value: string): void {
+    if (validateStringToBN(value)) {
+      this.root.tradeStore.closePositionMarketAmountOutBN = undefined
+      this.input =
+        value === this.selectedPosition?.totalValueInEth
+          ? { type: 'max' }
+          : { type: 'input', value }
+    }
+  }
+
+  get value(): string {
+    if (this.selectedMarket?.resolved || this.input.type === 'max') {
+      return this.selectedPosition?.totalValueInEth ?? ''
+    }
+
+    return this.input.value
+  }
+
+  get valueBN(): BigNumber | undefined {
+    return this.selectedPosition?.token.parseUnits(this.value)
+  }
```

Comment:
```suggestion
  get valueInEth(): string {
    if (this.selectedMarket?.resolved || this.input.type === 'max') {
      return this.selectedPosition?.totalValueInEth ?? ''
    }

    return this.input.value
  }

  get valueInEthBN(): BigNumber | undefined {
    return this.selectedPosition?.token.parseUnits(this.value)
  }
```

I think we should try to avoid having amount variables without specifying ETH/wstETH (I know we want to move to something like `value.inWstEth.bn` but still for the meantime)

Reply:
I would think this is pretty self explanatory tho since the input value is in ETH, but sure if it helps with readability

Reply:
Addressed this in this PR instead https://github.com/prepo-io/prepo-monorepo/pull/1225